CentralityiOS/CentralityiOS/ViewControllers/CategoryModalViewController.m

```diff
@@ -0,0 +1,124 @@
+        if (categories != nil) {
+            self.arrayOfCategories = [categories mutableCopy];
+        } else {
+            NSLog(@"%@", error.localizedDescription);
+        }
+        [self.categoryTableView reloadData];
+    }];
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section{
+    return self.arrayOfCategories.count;
+}
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
+{
+    CategoryObject *category = self.arrayOfCategories[indexPath.row];
+    CategoryCell *cell = [tableView dequeueReusableCellWithIdentifier:@"CategoryCell" forIndexPath:indexPath];
+    cell.category = category;
+    cell.categoryNameLabel.text = category.categoryName;
+    return cell;
+}
+
+- (void)tableView:(UITableView *)tableView
+didSelectRowAtIndexPath:(NSIndexPath *)indexPath{
+    CategoryObject *category = self.arrayOfCategories[indexPath.row];
+    [self.delegate didChangeCategory:category toFeed:self];
+    [self dismissViewControllerAnimated:YES completion:^{}];
+}
+
+- (UISwipeActionsConfiguration *)tableView:(UITableView *)tableView
+trailingSwipeActionsConfigurationForRowAtIndexPath:(NSIndexPath *)indexPath{
+    
+    UIContextualAction *deleteAction = [UIContextualAction contextualActionWithStyle:UIContextualActionStyleDestructive title:@"Delete" handler:^(UIContextualAction * _Nonnull action, __kindof UIView * _Nonnull sourceView, void (^ _Nonnull completionHandler)(BOOL)) {
+        
+        PFQuery *query = [self makeQuery];
+        
+        [query findObjectsInBackgroundWithBlock:^(NSArray *tasks, NSError *error) {
+            if (tasks != nil) {
+                [self.arrayOfCategories[indexPath.row] deleteInBackground];
+                [self.arrayOfCategories removeObjectAtIndex:indexPath.row];
+                [tableView deleteRowsAtIndexPaths:@[indexPath] withRowAnimation:UITableViewRowAnimationFade];
+                [self fetchCategories];
+            } else {
+                NSLog(@"%@", error.localizedDescription);
+            }
+        }];
+        
+        completionHandler(YES);
+        
+    }];
+    
+    deleteAction.backgroundColor = [UIColor systemRedColor];
+    
+    UISwipeActionsConfiguration *swipeActions = [UISwipeActionsConfiguration configurationWithActions:@[deleteAction]];
+    swipeActions.performsFirstActionWithFullSwipe=false;
+    return swipeActions;
+}
+
+- (IBAction)addCategoryAction:(id)sender {
+    if ([self.nameOfCategoryToAdd.text isEqual: @""]){
+        NSLog(@"Empty category name");
+        return;
+    }
+    
+    CategoryObject *newCategory = [CategoryObject new];
+    newCategory.categoryName = self.nameOfCategoryToAdd.text;
+    newCategory.owner = [PFUser currentUser];
+    
+    [newCategory saveInBackgroundWithBlock:^(BOOL succeeded, NSError * _Nullable error) {
+        if (succeeded) {
+            [self fetchCategories];
+            self.nameOfCategoryToAdd.text = @"";
```

Comment:
Why do we need to set it to an empty string?

Reply:
This is just preference. I figured it makes more sense to clear the field after the user adds a new category so they don't accidentally add the same category twice. Here's a demo of the feature : 
![Kapture 2022-07-21 at 15 35 54](https://user-images.githubusercontent.com/70764828/180326864-182eb5fb-afc1-4824-9f27-ef4d434d48a1.gif)